Source/sdc/sdc_newton_solve.H

```diff
@@ -312,6 +322,20 @@ sdc_newton_subdivide(const Real dt_m,
 
             sdc_newton_solve(dt_sub, U_begin, U_new, C, sdc_iteration, err_out, ierr);
 
+            if (ierr != newton::NEWTON_SUCCESS) {
+                // no point in continuing this subdivision if one stage failed
+                break;
+            }
+
+            // our solve may have resulted in mass fractions outside
+            // of [0, 1] -- reject if this is the case
+            for (int n = 0; n < NumSpec; ++n) {
+                if (U_new[UFS+n] < -newton::species_failure_tolerance * U_new[URHO] ||
+                    U_new[UFS+n] > (1.0_rt + newton::species_failure_tolerance) * U_new[URHO]) {
+                    ierr = newton::BAD_MASS_FRACTIONS;
```

Comment:
Does it makes sense to keep doing the sub-stepping if we hit this error case?

Reply:
Something like that, yes

Reply:
break added

Reply:
Well I'm not as worried about a break inside the species loop, I mean a break after the species loop before we finish up anything else we do below this

Reply:
ah, okay

Reply:
I just moved it to before the previous break.  Thanks.  I missed that.